hazelcast/include/hazelcast/util/Future.h

```diff
@@ -111,13 +193,28 @@ namespace hazelcast {
                 resultReady = false;
                 exceptionReady = false;
             }
-        private:
+
+            void andThen(const boost::shared_ptr<client::impl::ExecutionCallback<T> > &callback, util::Executor &executor) {
+                LockGuard guard(mutex);
+                callbacks.push_back(CallbackInfo(callback, executor));
```

Comment:
Implementation seems missing here. What if the response is already set to the future when andThen is called?

Reply:
Corrected and added tests.